axelor-contract/src/main/java/com/axelor/apps/contract/service/RevaluationFormulaService.java

```diff
@@ -0,0 +1,143 @@
+package com.axelor.apps.contract.service;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.contract.db.Contract;
+import com.axelor.apps.contract.db.ContractLine;
+import com.axelor.apps.contract.db.IndexRevaluation;
+import com.axelor.apps.contract.db.RevaluationFormula;
+import com.axelor.apps.contract.service.utils.FormulaProcessingResults;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+public class RevaluationFormulaService {
+  protected IndexRevaluationService indexRevaluationService;
+
+  @Inject
+  public RevaluationFormulaService(IndexRevaluationService indexRevaluationService) {
+    this.indexRevaluationService = indexRevaluationService;
+  }
+
+  public void checkFormula(RevaluationFormula revaluationFormula) throws ScriptException {
+    String formula = revaluationFormula.getFormula();
+    formula = formula.toLowerCase();
+    if (formula.contains("ind1i")) {
+      formula = formula.replace("ind1i", "1");
+    }
+    if (formula.contains("ind2i")) {
+      formula = formula.replace("ind2i", "1");
+    }
+    if (formula.contains("ind1f")) {
+      formula = formula.replace("ind1f", "1");
+    }
+    if (formula.contains("ind2f")) {
+      formula = formula.replace("ind2f", "1");
+    }
+    if (formula.contains("ind1py")) {
+      formula = formula.replace("ind1py", "1");
+    }
+    if (formula.contains("ind2py")) {
+      formula = formula.replace("ind2py", "1");
+    }
+    if (formula.contains("p0")) {
+      formula = formula.replace("p0", "1");
+    }
+    if (formula.contains("pf")) {
+      formula = formula.replace("pf", "1");
+    }
```

Comment:
it would be interesting to add all these constants into predefined variables, this way if we want to change the script syntax it would be easier.